packages/app/app/stacks/stacks-orders.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import {
+  StackOrder,
+  filterActiveOrders,
+  filterCompletedOrders,
+  getStackOrders,
+} from "@/models/stack-order";
+import { Tab } from "@headlessui/react";
+import { BodyText, ButtonLink, HeadingText } from "@/ui";
+import { StacksTable } from "@/components/StacksTable";
+import { EmptyState } from "@/app/stacks/empty-state";
+import { Order, getOrders } from "@/models/order";
+import { ChainId } from "@stackly/sdk";
+import { useEffect, useState } from "react";
+import { cx } from "class-variance-authority";
+
+export interface StackOrdersProps {
+  address: string;
+  chainId: ChainId;
+}
+
+export const StackOrders = ({ chainId, address }: StackOrdersProps) => {
+  const [loading, setLoading] = useState(true);
+  const [completedOrders, setCompletedOrders] = useState<StackOrder[]>([]);
+  const [activeOrders, setActiveOrders] = useState<StackOrder[]>([]);
+  const [currentOrders, setCurrentOrders] = useState<Order[]>([]);
+  const [currentStackOrders, setCurrentStackOrders] = useState<StackOrder[]>(
+    []
+  );
+
+  useEffect(() => {
+    async function fetchOrdersData() {
+      const orders = await getOrders(chainId, address.toLowerCase());
+      if (orders) setCurrentOrders(orders);
+    }
+    fetchOrdersData();
+  }, [address, chainId]);
+
+  useEffect(() => {
+    async function fetchStackOrdersData() {
+      const stackOrders = await getStackOrders(chainId, currentOrders);
+      if (stackOrders) setCurrentStackOrders(stackOrders);
+    }
+    fetchStackOrdersData();
+  }, [chainId, currentOrders]);
```

Comment:
I don't see a reason for saving the `currentOrders`. Maybe we can join this 2 fetches.

1. Fetch Orders
2. If orders, then fetch stackOrders
3. If no orders, stop loading
4. If stackOrders, then filter orders, set completed and active orders and stop loading
5. If no stackOrders, stop loading

In this case we just need to save active and completed orders, so less 2 `useState`.

Reply:
Yeah, probably we should do this. I'll fix that.